Mathlib/Combinatorics/Colex.lean

```diff
@@ -1,423 +1,414 @@
-
-theorem lt_trichotomy [LinearOrder α] (A B : Finset.Colex α) : A < B ∨ A = B ∨ B < A := by
-  by_cases h₁ : A = B
-  · tauto
-  have h : Finset.Nonempty (A \ B ∪ B \ A) := by
-    rw [nonempty_iff_ne_empty]
-    intro a
-    simp only [union_eq_empty, sdiff_eq_empty_iff_subset] at a
-    apply h₁ (Subset.antisymm a.1 a.2)
-  rcases exists_max_image (A \ B ∪ B \ A) id h with ⟨k, ⟨hk, z⟩⟩
-  · simp only [mem_union, mem_sdiff] at hk
-    cases' hk with hk hk
-    · right
-      right
-      refine' ⟨k, @fun t th => _, hk.2, hk.1⟩
-      specialize z t
-      by_contra h₂
-      simp only [mem_union, mem_sdiff, id.def] at z
-      rw [not_iff, iff_iff_and_or_not_and_not, not_not, and_comm] at h₂
-      apply not_le_of_lt th (z h₂)
-    · left
-      refine' ⟨k, @fun t th => _, hk.2, hk.1⟩
-      specialize z t
-      by_contra h₃
-      simp only [mem_union, mem_sdiff, id.def] at z
-      rw [not_iff, iff_iff_and_or_not_and_not, not_not, and_comm, or_comm] at h₃
-      apply not_le_of_lt th (z h₃)
-#align colex.lt_trichotomy Colex.lt_trichotomy
-
-instance [LinearOrder α] : IsTrichotomous (Finset.Colex α) (· < ·) :=
-  ⟨lt_trichotomy⟩
-
-instance decidableLt [LinearOrder α] : ∀ {A B : Finset.Colex α}, Decidable (A < B) :=
-  show ∀ {A B : Finset α}, Decidable (A.toColex < B.toColex) from @fun A B =>
-    decidable_of_iff' (∃ k ∈ B, (∀ x ∈ A ∪ B, k < x → (x ∈ A ↔ x ∈ B)) ∧ k ∉ A)
-      (by
-        rw [Colex.lt_def]
-        apply exists_congr
-        simp only [mem_union, exists_prop, or_imp, and_comm (a := _ ∈ B), and_assoc]
-        intro k
-        refine' and_congr_left' (forall_congr' _)
-        tauto)
-#align colex.decidable_lt Colex.decidableLt
-
-instance [LinearOrder α] : LinearOrder (Finset.Colex α) :=
-  { instLT,
-    instLE with
-    le_refl := fun A => Or.inr rfl
-    le_trans := le_trans
-    le_antisymm := fun A B AB BA =>
-      AB.elim (fun k => BA.elim (fun t => (asymm k t).elim) fun t => t.symm) id
-    le_total := fun A B =>
-      (lt_trichotomy A B).elim3 (Or.inl ∘ Or.inl) (Or.inl ∘ Or.inr) (Or.inr ∘ Or.inl)
-    -- Porting note: we must give some hints for instances
-    decidableLE := by
-      letI : DecidableEq (Finset.Colex α) := inferInstanceAs (DecidableEq (Finset α))
-      exact fun A B => inferInstanceAs (Decidable (A < B ∨ A = B))
-    decidableLT := inferInstance
-    decidableEq := inferInstanceAs (DecidableEq (Finset α))
-    lt_iff_le_not_le := fun A B => by
-      constructor
-      · intro t
-        refine' ⟨Or.inl t, _⟩
-        rintro (i | rfl)
-        · apply asymm_of _ t i
-        · apply irrefl _ t
-      rintro ⟨h₁ | rfl, h₂⟩
-      · apply h₁
-      apply h₂.elim (Or.inr rfl) }
-
-/-- The instances set up let us infer that `(· < ·)` is a strict total order. -/
-example [LinearOrder α] : IsStrictTotalOrder (Finset.Colex α) (· < ·) :=
-  inferInstance
+lemma toColex_image_le_toColex_image (hf : StrictMono f) :
+    toColex (s.image f) ≤ toColex (t.image f) ↔ toColex s ≤ toColex t := by
```

Comment:
Does this suggest we should make `StrictMono (fun s => s.image f)` (with some toColex scattered as appropriate) taking `StrictMono f` as an argument

Reply:
I've added the lemma but I'm not sure it's useful.